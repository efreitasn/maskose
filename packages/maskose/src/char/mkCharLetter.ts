import { MaskCharPredicateFn } from '.';

export const MASK_CHAR_LETTER_TYPE = 'MASK_CHAR_LETTER_TYPE';

export type MaskCharLetterObj = {
  type: typeof MASK_CHAR_LETTER_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

/**
 * A character in the range A to Z (case insensitive) expected to be in the value to be masked
 * @param predicateFn Function that will decide if the character will be present in the mask
 */
export default function mkCharLetter(predicateFn?: MaskCharPredicateFn): MaskCharLetterObj {
  return {
    type: MASK_CHAR_LETTER_TYPE,
    predicateFn
  };
}