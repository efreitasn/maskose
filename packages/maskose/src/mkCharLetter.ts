import { MaskCharPredicateFn } from './mkCharNum';

const MASK_CHAR_LETTER_TYPE = 'MASK_CHAR_LETTER_TYPE';

export type MkCharLetterObj = {
  type: typeof MASK_CHAR_LETTER_TYPE;
  predicateFn?: MaskCharPredicateFn;
};

export default function mkCharLetter(predicateFn?: MaskCharPredicateFn): MkCharLetterObj {
  return {
    type: MASK_CHAR_LETTER_TYPE,
    predicateFn
  };
}