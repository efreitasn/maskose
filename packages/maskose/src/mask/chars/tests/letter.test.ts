import mkCharLetter from '../letter';

test('should match the object returned by mkCharLetter()', () => {
  const result = mkCharLetter();

  expect(result).toMatchSnapshot();
});