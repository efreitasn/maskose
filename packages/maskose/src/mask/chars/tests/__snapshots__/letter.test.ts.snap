// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the object returned by mkCharLetter() 1`] = `
Object {
  "regExp": /\\[A-Z\\]/i,
  "repetitions": 1,
  "type": "MASKOSE_CHAR_LETTER_TYPE",
  "valueLengthConditions": Array [],
}
`;
